tests/models/components/test_seq2seq_encoder.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import torch
+
+from pie_models.models.components.seq2seq_encoder import (
+    ACTIVATION_TYPE2CLASS,
+    RNN_TYPE2CLASS,
+    build_seq2seq_encoder,
+)
+
+
+def test_no_encoder():
+    seq2seq_dict = {}
+    encoder, _ = build_seq2seq_encoder(seq2seq_dict, 10)
+    assert encoder is None
+
+    seq2seq_dict = {
+        "type": "sequential",
+        "rnn_layer": {
+            "type": "none",
+        },
+    }
+    encoder, _ = build_seq2seq_encoder(seq2seq_dict, 10)
+    assert len(encoder) == 0
+
+
+@pytest.mark.parametrize("seq2seq_enc_type", list(RNN_TYPE2CLASS))
+def test_rnn_encoder(seq2seq_enc_type):
+    seq2seq_dict = {
+        "type": "sequential",
```

Comment:
This should not be wrapped into `sequential` (also in tests below). Test all main if/elif branches of `build_seq2seq_encoder` in a separate test so that we can easily align any future test failure to the problematic code.